src/functions.rs

```diff
@@ -97,7 +97,7 @@ pub fn receive(channel: &UnixStream) -> Result<FdImplementor, String> {
     iov.iov_len = size_of::<c_int>();
 
     message.msg_control = controlbuf.as_mut_ptr() as *mut c_void;
-    message.msg_controllen = auto_cast!(size_of_val(&controlbuf));
+    message.msg_controllen = auto_cast!(controlbuf.len());
```

Comment:
I see,, given I've allocated a vec based on c_int size, size_of_val is irrelevant, where len does the job 🤔 

Thanks for identifying and offering a fix for this 👍